src/binding/python/RecordComponent.cpp

```diff
@@ -349,64 +357,83 @@ store_chunk(RecordComponent & r, py::array & a, py::tuple const & slices)
  * Size checks of the requested chunk (spanned data is in valid bounds)
  * will be performed at C++ API part in RecordComponent::loadChunk .
  */
-inline py::array
-load_chunk(RecordComponent & r, Offset const & offset, Extent const & extent, std::vector<bool> const & flatten)
+inline void
+load_chunk(RecordComponent & r, py::array & a, Offset const & offset, Extent const & extent)
 {
-    // some one-size dimensions might be flattended in our output due to selections by index
-    size_t const numFlattenDims = std::count(flatten.begin(), flatten.end(), true);
-    std::vector< ptrdiff_t > shape(extent.size() - numFlattenDims);
-    auto maskIt = flatten.begin();
-    std::copy_if(
-        std::begin(extent),
-        std::end(extent),
-        std::begin(shape),
-        [&maskIt](std::uint64_t){
-            return !*(maskIt++);
-        }
-    );
+    // check array is large enough
+    size_t s_load = 1u;
+    size_t s_array = 1u;
+    std::string str_extent_shape;
+    std::string str_array_shape;
+    for( auto & si : extent ) {
+	    s_load *= si;
+	    str_extent_shape.append(" ").append(std::to_string(si));
+    }
+    for( py::ssize_t d = 0; d < a.ndim(); ++d ) {
+	    s_array *= a.shape()[d];
+	    str_array_shape.append(" ").append(std::to_string(a.shape()[d]));
+    }
 
-    auto const dtype = dtype_to_numpy( r.getDatatype() );
+    /* we allow flattening of the result dimension
+    if( size_t(a.ndim()) != extent.size() )
```

Comment:
We could check this, but generally `<=` is allowed here
```suggestion
    if( size_t(a.ndim()) > extent.size() )
```